typed-protocols/src/Network/TypedProtocol/ReqResp/Server.hs

```diff
@@ -15,7 +15,7 @@ data ReqRespServer req resp m a = ReqRespServer {
 
     -- | The client terminated. Here we have a pure return value, but we
     -- could have done another action in 'm' if we wanted to.
-  , recvMsgDone  :: a
+  , recvMsgDone  :: m a
```

Comment:
There's nothing wrong with this of course (apart from inconsistency) but if we can solve the protocol thread bracket issue then perhaps we don't need this.

Reply:
I agree, it would be nicer to keep the protocols as simple as possible. I went with the least resistance at the first shot.